src/main/java/uk/gov/hmcts/reform/cwrdapi/service/impl/CaseWorkerServiceImpl.java

```diff
@@ -359,35 +370,93 @@ public void publishCaseWorkerDataToTopic(List<CaseWorkerProfile> caseWorkerData)
         return ResponseEntity.ok().body(mapCaseWorkerProfileToDto(caseWorkerProfileList));
     }
 
+    /**
+     * Returns the staff details for Refresh Assignments.
+     */
+    @Override
+    @SuppressWarnings("unchecked")
+    public ResponseEntity<Object> fetchStaffProfilesForRoleRefresh(String ccdServiceNames, PageRequest pageRequest) {
+        List<StaffProfileWithServiceName> staffProfileList;
+        Response lrdOrgInfoServiceResponse =
+                locationReferenceDataFeignClient.getLocationRefServiceMapping(ccdServiceNames);
+        ResponseEntity<Object> responseEntity = toResponseEntityWithListBody(lrdOrgInfoServiceResponse,
+                LrdOrgInfoServiceResponse.class);
+
+        if (responseEntity.getStatusCode().is2xxSuccessful()
+                && nonNull(responseEntity.getBody())
+                && responseEntity.getBody() instanceof List) {
+            List<LrdOrgInfoServiceResponse> listLrdServiceMapping =
+                    (List<LrdOrgInfoServiceResponse>)responseEntity.getBody();
+            if (CollectionUtils.isNotEmpty(listLrdServiceMapping)) {
+                Map<String, String> serviceNameToCodeMapping =
+                        listLrdServiceMapping
+                                .stream()
+                                .distinct()
+                                .filter(r -> StringUtils.isNotBlank(r.getServiceCode())
+                                        && StringUtils.isNotBlank(r.getCcdServiceName()))
+                                .collect(Collectors.toMap(LrdOrgInfoServiceResponse::getServiceCode,
+                                        LrdOrgInfoServiceResponse::getCcdServiceName));
+                Page<CaseWorkerWorkArea> workAreaPage = caseWorkerWorkAreaRepository.findByServiceCodeIn(
+                        serviceNameToCodeMapping.keySet(), pageRequest);
```

Comment:
can you please validate how much queries this fires specifically on case worker profile. if it fires single query fine. if it fires n no queries need to validate.

Reply:
sure, will do

Reply:
`Hibernate: select caseworker0_.case_worker_work_area_id as case_wor1_6_, caseworker0_.area_of_work as area_of_2_6_, caseworker0_.case_worker_id as case_wor3_6_, caseworker0_.created_date as created_4_6_, caseworker0_.last_update as last_upd5_6_, caseworker0_.service_code as service_6_6_ from case_worker_work_area caseworker0_ where caseworker0_.service_code in (?) order by caseworker0_.case_worker_id asc limit ?
Hibernate: select count(caseworker0_.case_worker_work_area_id) as col_0_0_ from case_worker_work_area caseworker0_ where caseworker0_.service_code in (?)`

only these queries are being fired at this point. as we have "FetchType as lazy" we get the case worker profile data only at the point it has been invoked